core/cloudflow-flink/src/main/scala/cloudflow/flink/FlinkStreamletContextImpl.scala

```diff
@@ -46,7 +47,7 @@ class FlinkStreamletContextImpl(
 
     val properties = new ju.Properties
     properties.setProperty("bootstrap.servers", config.getString("cloudflow.kafka.bootstrap-servers"))
-    properties.setProperty("group.id", srcTopic)
+    properties.setProperty("group.id", groupId(inlet))
```

Comment:
@debasishg srcTopic, I think resolves to `SavepointPath.value`, which is  s"${appId}.${streamletRef}.${portName}".
`SavepointPath` encodes this. Maybe we can reuse that? and also fix this in the `AkkaStreamletContextImpl`? (I think the groupId function there takes a streamletRef and inlet that is not necessary)
So what I mean is move the code to create groupId to SavepointPath, since this must encode what is needed for connecting to the savepoint (the topic).

Reply:
maybe resolvePort can also be moved to StreamletContext so it works for all streamlets.

Reply:
@ray I like this that u did in Akka ..

```scala
private def findSavepointPathForPort(port: StreamletPort): SavepointPath =
  streamletDefinition
    .resolveSavepoint(port)
    .getOrElse(throw PortNotFoundException(port, this))
```

better than `resolvePort` returning an opaque string. And the following is much more meaningful ..

```scala
val savepointPath = findSavepointPathForPort(inlet)
val topic         = savepointPath.value
```

I will remove `resolvePort` and move `findSavepointPathForPort` to `StreamletContext` so that it can be used as above from all runtimes.

Agree?

Reply:
yes